packages/testing/src/execution_testing/cli/pytest_commands/plugins/shared/benchmarking.py

```diff
@@ -54,103 +56,211 @@ def pytest_configure(config: pytest.Config) -> None:
-        if pattern == test_name:
-            continue
-        try:
-            if re.search(pattern, test_name):
-                return counts
-        except re.error:
-            continue
+    @classmethod
+    @abstractmethod
+    def from_parameter_value(
+        cls, config: pytest.Config, value: str
+    ) -> Self | None:
+        """Given the parameter value and config, return the expected object."""
+        pass
+
+    @classmethod
+    def from_config(cls, config: pytest.Config) -> Self | None:
+        """
+        Parse the config the parametrizer configuration from the config.
+
+        Return `None` if the parameter is not specified.
+        """
+        if hasattr(config, cls.config_field):
+            return getattr(config, cls.config_field)
+
+        parameter_value = config.getoption(cls.parameter_name)
+        setattr(config, cls.config_field, None)
 
-    return default_counts
+        if parameter_value is None:
+            return None
+        else:
+            setattr(
+                config,
+                cls.config_field,
+                cls.from_parameter_value(config, parameter_value),
+            )
+        return getattr(config, cls.config_field)
+
+    @abstractmethod
+    def get_test_parameters(self, test_name: str) -> list[ParameterSet]:
+        """Get the parameters list for a given test."""
+        pass
+
+    def parametrize(self, metafunc: pytest.Metafunc) -> None:
+        """Parametrize a test."""
+        if self.parameter_name in metafunc.fixturenames:
+            test_name = metafunc.function.__name__
+            metafunc.parametrize(
+                self.parameter_name,
+                self.get_test_parameters(test_name),
+                scope="function",
+            )
+
+
+class GasBenchmarkValues(RootModel, BenchmarkParametrizer):
+    """Gas benchmark values configuration object."""
+
+    root: List[int]
+
+    flag: ClassVar[str] = "--gas-benchmark-values"
+    config_field: ClassVar[str] = "_gas_benchmark_values_config"
+    parameter_name: ClassVar[str] = "gas_benchmark_value"
+
+    @classmethod
+    def from_parameter_value(
+        cls, config: pytest.Config, value: str
+    ) -> Self | None:
+        """Given the parameter value and config, return the expected object."""
+        return cls.model_validate(value.split(","))
+
+    def get_test_parameters(self, test_name: str) -> list[ParameterSet]:
+        """Get benchmark values. All tests have the same list."""
+        return [
+            pytest.param(
+                gas_value * 1_000_000,
+                id=f"benchmark-gas-value_{gas_value}M",
+            )
+            for gas_value in self.root
+        ]
+
+
+class OpcodeCountsConfig(BaseModel, BenchmarkParametrizer):
+    """Opcode counts configuration object."""
+
+    scenario_configs: Dict[str, List[int]] = Field(default_factory=dict)
+    default_counts: List[int] = Field(default_factory=lambda _: [1])
```

Comment:
This lambda doesn't take a positional argument. I think pydantic introspects and passes a value (which avoids a runtime `TypeError`). But I think it's cleaner to omit it.

```suggestion
    default_counts: List[int] = Field(default_factory=lambda : [1])
```